repositories/professor_repository.go

```diff
@@ -0,0 +1,126 @@
+package repositories
+
+import (
+	"class-review-backend/models"
+	"log"
+
+	"github.com/jmoiron/sqlx"
+)
+
+type IProfessorRepository interface {
+	GetAllProfessors() (*[]models.Professor, error)
+	GetProfessorByUniqname(string)(*models.Professor, error)
+	PostProfessor(*models.Professor) error
+	UpdateProfessor(*models.Professor, string) (*models.Professor, error)
+	DeleteProfessor(string) error
+	GetProfessorStats() (*[]models.ProfessorStats, error)
+	GetProfessorStatsByUniqname(string) (*models.ProfessorStats, error)
+}
+
+// Implements IProfessorRepository
+type ProfessorRepository struct {
+	Database *sqlx.DB
+}
+
+func DefaultProfessorRepository(db *sqlx.DB) *ProfessorRepository {
+	return &ProfessorRepository{
+		Database: db,
+	}
+}
+
+func (pr *ProfessorRepository) GetAllProfessors() (*[]models.Professor, error) {
```

Comment:
You don't need pointers-to-slices, since slices already act as pointers to the underlying data